mobileapps/views.py

```diff
@@ -296,3 +302,187 @@ def delete(self, request, mobileapp_id):
         except ObjectDoesNotExist:
             raise Http404
 
+
+class MobileAppAllUsersNotifications(SecureAPIView):
+    """
+    **Use Cases**
+
+        send a push notification to all the users of the specific app
+        using app's push notifications provider.
+
+    **Example Requests**
+
+        POST /api/mobileapps/{id}/notification
+
+        The body of the POST request must include the following parameters.
+
+        * message: notification message
+
+    **Response Values**
+
+        If the request is successful, the request returns an HTTP 202 "Accepted" response.
+
+        The HTTP 202 response has the following value.
+
+        * message: Accepted
+    """
+
+    def post(self, request, pk):
+
+        message = request.data.get('message', None)
+        if not message:
+            return Response({'message': _('message is missing')}, status.HTTP_400_BAD_REQUEST)
+
+        try:
+            mobile_app = MobileApp.objects.get(pk=pk)
+            if not mobile_app.is_active:
+                return Response({'message': _('Mobile app is inactive')}, status.HTTP_403_FORBIDDEN)
+        except ObjectDoesNotExist:
+            return Response({'message': _('Mobile app does not exist')}, status.HTTP_404_NOT_FOUND)
+
+        try:
+            api_keys = mobile_app.get_api_keys()
+            notification_provider = mobile_app.get_notifications_provider()
+            if not notification_provider:
```

Comment:
Can we move this block of code little up where we are checking if mobile app is active?.